tpm_test.go

```diff
@@ -804,10 +823,21 @@ func TestMain(m *testing.M) {
 				testCAKey = key
 			}
 
-			tcti, err := tpm2.OpenMssim(*mssimHost, *mssimTpmPort, *mssimPlatformPort)
-			if err != nil {
-				fmt.Fprintf(os.Stderr, "Failed to open mssim connection: %v", err)
-				return 1
+			var tcti *tpm2.TctiMssim
+			// Give the simulator 5 seconds to start up
+		Loop:
+			for i := 0; ; i++ {
+				var err error
+				tcti, err = tpm2.OpenMssim("", *mssimPort, *mssimPort+1)
+				switch {
+				case err != nil && i == 4:
```

Comment:
Is there a real situation where the simulator takes several seconds to start, or we're just playing it safe?

Reply:
Not several seconds, but this code does race with the simulator startup without having a retry here.